ui/src/app/components/new-user-satisfaction-survey-modal.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import { useState } from 'react';
+
+import { NewUserSatisfactionSurveySatisfaction } from 'generated/fetch';
+
+import { Button } from 'app/components/buttons';
+import { FlexColumn, FlexRow } from 'app/components/flex';
+import {
+  ErrorMessage,
+  TextAreaWithLengthValidationMessage,
+} from 'app/components/inputs';
+import { Modal } from 'app/components/modals';
+import { MultipleChoiceQuestion } from 'app/components/multiple-choice-question';
+import { surveysApi } from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import {
+  createNewUserSatisfactionSurveyStore,
+  useStore,
+} from 'app/utils/stores';
+
+const orderedSatisfactionOptions = [
+  {
+    value: NewUserSatisfactionSurveySatisfaction.VERYUNSATISFIED,
+    label: 'Very unsatisfied',
+  },
+  {
+    value: NewUserSatisfactionSurveySatisfaction.UNSATISFIED,
+    label: 'Unsatisfied',
+  },
+  {
+    value: NewUserSatisfactionSurveySatisfaction.NEUTRAL,
+    label: 'Neutral',
+  },
+  {
+    value: NewUserSatisfactionSurveySatisfaction.SATISFIED,
+    label: 'Satisfied',
+  },
+  {
+    value: NewUserSatisfactionSurveySatisfaction.VERYSATISFIED,
+    label: 'Very satisfied',
+  },
+];
+
+export const ADDITIONAL_INFO_MAX_CHARACTERS = 500;
+
+export interface NewUserSatisfactionSurveyModalProps {
+  onCancel: () => void;
+  onSubmitSuccess: () => void;
+}
+
+export function NewUserSatisfactionSurveyModal({
+  onCancel,
+  onSubmitSuccess,
+}: NewUserSatisfactionSurveyModalProps) {
+  const { newUserSatisfactionSurveyData } = useStore(
+    createNewUserSatisfactionSurveyStore
+  );
+  const [submittingRequest, setSubmittingRequest] = useState(false);
+  const [error, setError] = useState(false);
+
+  return (
+    <Modal width={850} onRequestClose={onCancel}>
+      <FlexColumn style={{ gap: '0.5rem' }}>
+        <MultipleChoiceQuestion
+          question='How would you rate your overall satisfaction with the Researcher Workbench?'
+          options={orderedSatisfactionOptions}
+          selected={newUserSatisfactionSurveyData.satisfaction}
+          onChange={(satisfaction: NewUserSatisfactionSurveySatisfaction) => {
+            createNewUserSatisfactionSurveyStore.set({
+              newUserSatisfactionSurveyData: {
+                ...newUserSatisfactionSurveyData,
+                satisfaction: satisfaction,
+              },
+            });
+          }}
+        />
+        <label
+          htmlFor='new-user-satisfaction-survey-additional-info'
+          style={{
+            fontWeight: 'bold',
+            color: colors.primary,
+            fontSize: '18px',
+            lineHeight: '1.25rem',
+          }}
```

Comment:
This is copied directly from the MultipleChoiceQuestion question style. I'm open to suggestions for a better way to do this.

Reply:
`surveyQuestionStyle` in a new `components/surveys/common.tsx` file?

Reply:
or `components/surveys.tsx`